inc/Engine/Media/AboveTheFold/AJAX/Controller.php

```diff
@@ -248,4 +262,60 @@ public function check_lcp_data() {
 
 		wp_send_json_error( 'data does not exist' );
 	}
+
+	/**
+	 * Validate image object.
+	 *
+	 * @param object $image_object Image full object.
+	 * @return bool
+	 */
+	private function validate_image( $image_object ) {
+		$valid_image = ! empty( $image_object->src ) ? $this->validate_image_src( $image_object->src ?? '' ) : true;
+
+		/**
+		 * Filters If the image src is a valid image or not.
+		 *
+		 * @param bool   $valid_image Valid image or not.
+		 * @param string $image_src_url Image src url.
+		 * @param object $image_object Image object with full details.
+		 */
+		return (bool) apply_filters( 'rocket_atf_valid_image', $valid_image, $image_object->src, $image_object );
+	}
+
+	/**
+	 * Make sure that this url is valid image without loading the image itself.
+	 *
+	 * @param string $image_src Image src url.
+	 * @return bool
+	 */
+	private function validate_image_src( $image_src ) {
+		if ( empty( $image_src ) ) {
+			return false;
+		}
+
+		/**
+		 * Filters the supported schemes of LCP/ATF images.
+		 *
+		 * @param array  $invalid_schemes Array of invalid schemes.
+		 */
+		$invalid_schemes = apply_filters( 'rocket_atf_invalid_schemes', $this->invalid_schemes );
+		$invalid_schemes = implode( '|', $invalid_schemes );
+
+		if ( preg_match( '#^' . $invalid_schemes . '#', $image_src ) ) {
+			return false;
+		}
+
+		// Here we get the url PATH part only to strip all query strings.
+		$image_src_path = wp_parse_url( $image_src, PHP_URL_PATH );
+		if ( empty( $image_src_path ) ) {
+			return false;
+		}
+
+		// Add svg to allowed mime types.
+		$allowed_mime_types        = get_allowed_mime_types();
+		$allowed_mime_types['svg'] = 'image/svg+xml';
+		$image_src_filetype_array  = wp_check_filetype( $image_src_path, $allowed_mime_types );
```

Comment:
```suggestion
		/**
		 * Filters the supported schemes of LCP/ATF images.
		 *
		 * @param array  $invalid_schemes Array of invalid schemes.
		 */
		$invalid_schemes = apply_filters( 'rocket_atf_invalid_schemes', array( 'chrome-extension' ) );

		$image_src_components = wp_parse_url( $image_src );

		if ( empty( $image_src_components['path'] ) || ( ! empty( $image_src_components['scheme'] ) && in_array( strtolower( $image_src_components['scheme'] ), $invalid_schemes, true ) ) ) {
			return false;
		}

		// Add svg to allowed mime types.
		$allowed_mime_types        = get_allowed_mime_types();
		$allowed_mime_types['svg'] = 'image/svg+xml';
		$image_src_filetype_array  = wp_check_filetype( $image_src_components['path'], $allowed_mime_types );
```

This is what I meant, what do u think?

Reply:
@wordpressfan This is fine, but I prefer the flexibility that comes with regex as we could have `chrome-extensions:// `but we could also have `chrome-untrusted://` in this case, we would have to try to capture all kinds of `chrome-*` in the list, this is the reason I went this way.

Reply:
As you like, I just hate regex :D because it can throw unexpected failures regarding some inputs or it may take time more than in_array but I'm sure it'll do the trick.

Can you plz just guard the filter return to be an array and not empty?